### DISCUSSION TITLE
If I can obtain the intermediate generated HLS file?

### DISCUSSION THREAD

**Discussion by sansi-zhang** [Xilinx/finn#880] [Q&A] [answered]
If I want to get the HLS file of my network model, can I get it from the finn framework function or something?

If possible, I hope you can give me some specific operation suggestions or guidance.

Thanks.

**Comment by fpjentzsch** [comment, marked as answer]
Hi,
after HLS codegen/ipgen, the HLS CustomOP nodes in the model will have a `code_gen_dir_ipgen` attribute pointing to the Vitis HLS project generated by FINN. It will be located within your FINN_HOST_BUILD_DIR directory.

**Reply by sansi-zhang** [reply]
@fpjentzsch 
Hi, I'm sorry to bother you again.

I tried again to get and understand the hls code, but again I failed.

I did not look for the node information you mentioned in my onnx file. And I tried to look at each MVU and VVU file to get the hls code, but I found that it seemed to be incomplete.

In this regard, I ask for your help and hope that you can give me some suggestions and measures.

**Comment by sansi-zhang** [comment]
But in that directory, I only found folders corresponding to each node of the onnx file, not files containing calls to the entire process. And I tried to look at all the generated folders, but I couldn't understand the patterns and the relationship between the various files.
Below is the file I generated earlier.
![image](https://github.com/Xilinx/finn/assets/85163357/c35a2ecc-b45b-4c21-bc82-8f39fa59d855)

I want to get the full HLS code and make specific changes to it. If possible, I hope you can give me some guidance and suggestions.


Thanks.

**Reply by hleblevec** [reply]
Hi,

The HLS code of every node in the model is generated and synthesized out-of-context, and then every generated IP is stitched together to form the architecture. As far as I know, there isn't any global HLS code that would describe the whole model. Here, every folder contains the generated HLS code and a Vitis HLS project containing (amongst other things) the IP files for each individual node in the ONNX model.

**Comment by sansi-zhang** [comment]
Thank you very much. I understand.

---- Replied Message ----
| From | Hugo LE ***@***.***> |
| Date | 08/31/2023 19:12 |
| To | ***@***.***> |
| Cc | zhangsansi ***@***.***>,
Author ***@***.***> |
| Subject | Re: [Xilinx/finn] If I can obtain the intermediate generated HLS file? (Discussion #880) |

Hi,

The HLS code of every node in the model is generated and synthesized out-of-context, and then every generated IP is stitched together to form the architecture. As far as I know, there isn't any global HLS code that would describe the whole model. Here, every folder contains the generated HLS code and a Vitis HLS project containing (amongst other things) the IP files for each individual node in the ONNX model.
